site/src/components/service-message-list.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  ServiceDefinition,
+  ServiceMessageDefinition,
+} from "@blockprotocol/core";
+import { Box, Typography } from "@mui/material";
+import * as React from "react";
+import slugify from "slugify";
+
+import { JsonSchema } from "../lib/json-schema";
+import { mdxComponents } from "../util/mdx-components";
+
+const DataType: React.VoidFunctionComponent<{ propertySchema: JsonSchema }> = ({
+  propertySchema,
+}) => {
+  const MdxA = mdxComponents.a!;
+
+  const isArray = propertySchema.type === "array";
+  const { type: innerType, $ref } = isArray
+    ? (propertySchema.items as JsonSchema)
+    : propertySchema;
+
+  return $ref ? (
+    <MdxA href={$ref} target="_blank">
+      {$ref.match(/[^/]+(?=\/$|$)/)?.[0]}
+      {isArray ? "[]" : ""}
+    </MdxA>
```

Comment:
I might be misremembering, but I think our link component opens external URLs in a new tab automatically.
```suggestion
    <MdxA href={$ref}>
      {$ref.match(/[^/]+(?=\/$|$)/)?.[0]}
      {isArray ? "[]" : ""}
    </MdxA>
```

Reply:
The Link component automatically converts anything containing `blockprotocol.org` into the relative format. I think I'll leave it doing this since these are a special case

Reply:
Ah good point, I didn't consider that the references start with `blockprotocol.org`